modules/bitgo/test/v2/unit/tss/ecdsa.ts

```diff
@@ -380,6 +388,169 @@ describe('Ecdsa tss helper functions tests', function () {
+        dShare.i.should.equal(mockDShareToBitgo.i);
+        dShare.j.should.equal(mockDShareToBitgo.j);
+        dShare.delta.should.equal(mockDShareToBitgo.delta);
+        dShare.Gamma.should.equal(mockDShareToBitgo.Gamma);
+      });
+  
+      it('should successfully convert D share to signarture share record', function () {
+        const share = {
+          to: SignatureShareType.BITGO,
+          from: SignatureShareType.USER,
+          share: mockDShareToBitgo.delta + mockDShareToBitgo.Gamma,
+        } as SignatureShareRecord;
+        const dShare = ECDSAMethods.convertDShare(mockDShareToBitgo);
+        dShare.from.should.equal(share.from);
+        dShare.to.should.equal(share.to);
+        dShare.share.should.equal(share.share);
+      });
+
+      it('should successfully parse S and D share', function() {
+        const share = {
+          to: SignatureShareType.BITGO,
+          from: SignatureShareType.USER,
+          share: mockSignature.r + mockSignature.s + mockSignature.y + mockDShareToBitgo.delta + mockDShareToBitgo.Gamma,
+        } as SignatureShareRecord;
+        const { sShare, dShare } = ECDSAMethods.parseSDShare(share);
+        sShare.i.should.equal(3);
+        sShare.r.should.equal(mockSignature.r);
+        sShare.s.should.equal(mockSignature.s);
+        sShare.y.should.equal(mockSignature.y);
+  
+        dShare.i.should.equal(mockDShareToBitgo.i);
+        dShare.j.should.equal(mockDShareToBitgo.j);
+        dShare.delta.should.equal(mockDShareToBitgo.delta);
+        dShare.Gamma.should.equal(mockDShareToBitgo.Gamma);
+      });
+
+      it('should successfully convert S and D share to signarture share record', function () {
+        const share = {
+          to: SignatureShareType.BITGO,
+          from: SignatureShareType.USER,
+          share: mockSignature.r + mockSignature.s + mockSignature.y + mockDShareToBitgo.delta + mockDShareToBitgo.Gamma,
+        } as SignatureShareRecord;
+        const sdShare = ECDSAMethods.convertSDShare({ sShare: mockSignature, dShare: mockDShareToBitgo });
+        sdShare.from.should.equal(share.from);
+        sdShare.to.should.equal(share.to);
+        sdShare.share.should.equal(share.share);
+      });
+  
+      it('should successfully parse signature share', function() {
+        const share = {
+          to: SignatureShareType.USER,
+          from: SignatureShareType.BITGO,
+          share: mockSignature.r + mockSignature.s + mockSignature.y,
+        } as SignatureShareRecord;
+        const signature = ECDSAMethods.parseSignatureShare(share);
+        signature.i.should.equal(1);
+        signature.r.should.equal(mockSignature.r);
+        signature.s.should.equal(mockSignature.s);
+        signature.y.should.equal(mockSignature.y);
+      });
+
+      it('should successfully convert signature share to signarture share record', function () {
+        const share = {
+          to: SignatureShareType.USER,
```

Comment:
NIT: Lets add a test for the error cases also

Reply:
I will have follow up pr with this kind of test